File: java/0007_Reverse_Integer.java
class Solution {
    public int reverse(int x) {
        boolean isNegative = x < 0;
        long xl = Math.abs(x);
        long res = 0;

        while (xl > 0) {
            res = res * 10 + xl % 10;
            xl /= 10;
            if (res > Integer.MAX_VALUE) {
                return 0;
            }
        }

        return (int) (isNegative ? -res : res);
    }
}

File: kotlin/0007_Reverse_Integer.kt
class Solution {
    fun reverse(x: Int): Int {
        val str = x.toString()
        val res = if (str[0] != '-') {
            str.reversed().toLong()
        } else {
            str.substring(1).reversed().toLong() * -1
        }

        return if (Int.MIN_VALUE < res && res < Int.MAX_VALUE) res.toInt() else 0
    }
}
